igor-web/src/main/groovy/com/netflix/spinnaker/igor/model/ScmDetails.groovy

```diff
@@ -32,6 +32,13 @@ import org.simpleframework.xml.Root
 class ScmDetails {
     @Element(required = false)
     Action action
+
+    GenericGitRevision genericGitRevision() {
+        if (action?.lastBuiltRevision?.branch?.name) {
+            return new GenericGitRevision(action.lastBuiltRevision.branch.first().name, action.lastBuiltRevision.branch.first().name.split('/').last(), action.lastBuiltRevision.branch.first().sha1)
```

Comment:
This line here: https://github.com/spinnaker/igor/pull/77/files#diff-cdf1a482044f79d54f93d79181d4b209L64 seems to modify everything in the `branch` attribute, but here you only peel off the `first()` object in `branch`. Are you sure the behavior is identical before and after your changes?